code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -700,21 +700,23 @@ var/global/list/obj/item/device/pda/PDAs = list()
 	set name = "Remove Pen"
 	set src in usr
 
+	var/obj/item/inserted_item = loc
+
 	if(issilicon(usr))
 		return
 
+	var/i
 	if (usr.canUseTopic(src))
-		var/obj/item/weapon/pen/O = locate() in src
-		if(O)
-			if (istype(loc, /mob))
-				var/mob/M = loc
-				if(M.get_active_hand() == null)
-					M.put_in_hands(O)
-					usr << "<span class='notice'>You remove \the [O] from \the [src].</span>"
+		for(i = 1, i <= contained_item.len, i++)
+			inserted_item = locate(contained_item[i]) in src	//If items are in the PDA

```

Comment:
I told you to make inserted_item a var for the PDA. That way the code here can be really simplified since we don't have to search for an item.